core/meta/inc/TIsAProxy.h

```diff
@@ -28,14 +28,22 @@ class TIsAProxy  : public TVirtualIsAProxy {
 private:
    template <typename T> using Atomic_t = std::atomic<T>;
 
+   // On testing with the data from the 250202_181_RECO.root and doing "just" serializing
+   // the value 8 was the sweet spot of performance.  With more slots, too much time is
+   // spent scanning the array of "last" seen and with less slots then the
+   // serialization induced by and/or the cost of executed `++fSubTypesReaders is slow
+   // down (noticeably) the streaming of branches with polymorphic containers.
+   static constexpr UInt_t fgMaxLastSlot = 8;
```

Comment:
The bottleneck is not (as far as I can tell) the load of the atomic but rather "simply" whether the value needs to be recalculated or not (i.e. the calculation requires taking either a localized read/write lock or the global read/write lock but of which even if only the read lock is taken are reducing parallelism (the bottleneck was the increment of the count of the number of outstanding reader).

Reply:
I just mean that this can cause false sharing if the slots are frequently read and updated, which would degrade performance. It's something that `perf c2c` can help to clarify or rule out if needed.